samples/sample-chat-app/src/main/java/com/azure/android/communication/chat/sampleapp/MainActivity.java

```diff
@@ -82,12 +93,27 @@
     private String threadId = "<to-be-updated-below>";
     private String chatMessageId = "<to-be-updated-below>";
     private final String endpoint = "";
-    private final String sdkVersion = "1.0.0";
+    private final String sdkVersion = "1.1.0-beta.2";
     private static final String SDK_NAME = "azure-communication-com.azure.android.communication.chat";
-    private static final String APPLICATION_ID = "Chat Test App";
+    private static final String APPLICATION_ID = "Chat_Test_App";
     private static final String TAG = "[Chat Test App]";
     private final Queue<String> unreadMessages = new ConcurrentLinkedQueue<>();
-    private static RealTimeNotificationCallback messageReceivedHandler;
+    private static Map<RealTimeNotificationCallback, ChatEventType> realTimeNotificationCallbacks = new HashMap<>();
+
+    private BroadcastReceiver firebaseMessagingReceiver = new BroadcastReceiver() {
+        @Override
+        public void onReceive(Context context, Intent intent) {
+            ChatPushNotification pushNotification =
+                (ChatPushNotification) intent.getParcelableExtra("PushNotificationPayload");
+
+            Log.d(TAG, "Push Notification received in MainActivity: " + pushNotification.getPayload());
+
+            if (chatAsyncClient != null) {
+                Log.d(TAG, "Passing push notification to chatAsyncClient.");
+                chatAsyncClient.handlePushNotification(pushNotification);
+            }
```

Comment:
Is there any scenario where the client could be `null` but this object isn't?

Reply:
Actually, no. We would call "createChatAsyncClient()" before register the receiver in "onCreate". I think I can remove this check.